### PR TITLE
ui: Keep the tasks sorted

### PR DESCRIPTION
**Pull request by aleb**

### CODE REVIEW

File: ui/src/actions/planning.js [hunk 1]
```diff
@@ -19,7 +19,26 @@
 
 export const PLAN_SET = 'PLAN_SET';
 
+function byDateAgeFn(a, b) {
+  // Order by start date.
+  // Replace "" with "Z" so the tasks with no starting date appear below.
+  var a_start_date = a.start_date || "Z";
```
Comment (ljanyst) [inline ui/src/actions/planning.js]: Let's keep the camel case whenever possible for consistency.

File: ui/src/components/Gantt.js [hunk 2]
```diff
@@ -315,14 +307,13 @@ class Gantt extends Component {
...
-    gantt.sort("text", false);
-    gantt.render();
-    gantt.scrollTo(this.scrollPos.x, this.scrollPos.y);
-  }
-
   render() {
+    this.scrollPos = gantt.getScrollState();
+    this.expandedTasks = new Map();
+    gantt.eachTask(task => {
+      this.expandedTasks.set(task.id, task.$open);
```
Comment (ljanyst) [inline ui/src/components/Gantt.js:314]: Does that work when you switch between projects?

Comment (aleb) [reply ui/src/components/Gantt.js:314]: This `eachTask` works fine, as it only gathers data. But the part below should leave the task open if it's not in the map, let me fix it.

**Comment by ljanyst** [comment]
LGTM. Thanks Alex.

**Pull request merged**